source/OutgoingMessages.IntegrationTests/Application/UseCases/DelegateMessageTests.cs

```diff
@@ -0,0 +1,136 @@
+using Energinet.DataHub.EDI.OutgoingMessages.Domain.Models.ActorMessagesQueues;
+using Energinet.DataHub.EDI.OutgoingMessages.Domain.Models.OutgoingMessages;
+using Energinet.DataHub.EDI.OutgoingMessages.IntegrationTests.Fixtures;
+using NodaTime;
+using Xunit.Abstractions;
+
+namespace Energinet.DataHub.EDI.OutgoingMessages.IntegrationTests.Application.UseCases;
+
+public class DelegateMessageTests : OutgoingMessagesTestBase
+{
+    private const string GridAreaCode = "001";
+    private readonly Actor _delegatedTo;
+    private readonly Actor _delegatedBy;
+    private readonly DelegateMessage _sut;
+    private readonly Instant _now;
+
+    public DelegateMessageTests(
+        OutgoingMessagesTestFixture outgoingMessagesTestFixture,
+        ITestOutputHelper testOutputHelper)
+        : base(outgoingMessagesTestFixture, testOutputHelper)
+    {
+        _delegatedTo = new Actor(ActorNumber.Create("1111111111111"), actorRole: ActorRole.Delegated);
+        _delegatedBy = new Actor(ActorNumber.Create("4444444444444"), actorRole: ActorRole.EnergySupplier);
+
+        _sut = GetService<DelegateMessage>();
+        _now = SystemClock.Instance.GetCurrentInstant();
+    }
+
+    public static TheoryData<ProcessType> GetAllDelegationProcessTypes => new(
+        EnumerationType.GetAll<ProcessType>()
+            .Where(x => !NotDelegationProcessTypes().Contains(x))
+            .ToArray());
+
+    public static IEnumerable<ProcessType> NotDelegationProcessTypes()
+    {
+        return new List<ProcessType>
+        {
+            ProcessType.RequestEnergyResults,
+            ProcessType.RequestWholesaleResults,
+        };
+    }
+
+    [Theory]
+    [MemberData(nameof(GetAllDelegationProcessTypes))]
+    public async Task Given_DelegationIsSet_When_DelegateAsync_Then_MessageIsDelegated(ProcessType processType)
+    {
+        var message = CreateOutgoingMessage(_delegatedBy, processType);
+        await AddDelegationAsync(_delegatedBy, _delegatedTo, processType);
+
+        var result = await _sut.DelegateAsync(message, CancellationToken.None);
+
+        Assert.Equal(_delegatedBy.ActorNumber, result.DocumentReceiver.Number);
+        Assert.Equal(_delegatedBy.ActorRole, result.DocumentReceiver.ActorRole);
+
+        Assert.Equal(_delegatedTo.ActorNumber, result.Receiver.Number);
+        Assert.Equal(_delegatedTo.ActorRole, result.Receiver.ActorRole);
+    }
+
+    [Theory]
+    [MemberData(nameof(GetAllDelegationProcessTypes))]
+    public async Task Given_DocumentTypeIsAcknowledgement_When_DelegateAsync_Then_NoMessagesAreDelegated(ProcessType processType)
+    {
+        var message = CreateOutgoingMessage(_delegatedBy, processType, DocumentType.Acknowledgement);
+        await AddDelegationAsync(_delegatedBy, _delegatedTo, processType);
+
+        var result = await _sut.DelegateAsync(message, CancellationToken.None);
+
+        Assert.Equal(_delegatedBy.ActorNumber, result.DocumentReceiver.Number);
```

Comment:
```suggestion
        Assert.Equal(result.DocumentReceiver.Number, _delegatedBy.ActorNumber);
```

expected is the first parameter of Equeal()